config/initializers/require_listeners.rb

```diff
@@ -27,39 +27,4 @@
 # See doc/COPYRIGHT.rdoc for more details.
 #++
 
-class WorkPackageObserver < ActiveRecord::Observer
-  attr_accessor :send_notification
-
-  def after_create(work_package)
-    if send_notification
-      recipients = work_package.recipients + work_package.watcher_recipients
-      users = User.find_all_by_mails(recipients.uniq)
-
-      users.each do |user|
-        notify(user, work_package)
-      end
-    end
-    clear_notification
-  end
-
-  ##
-  # Notifies the user of the created work package.
-  def notify(user, work_package)
-    job = DeliverWorkPackageCreatedJob.new(user.id, work_package.id, User.current.id)
-
-    Delayed::Job.enqueue job
-  end
-
-  # Wrap send_notification so it defaults to true, when it's nil
-  def send_notification
-    return true if @send_notification.nil?
-    @send_notification
-  end
-
-  private
-
-  # Need to clear the notification setting after each usage otherwise it might be cached
-  def clear_notification
-    @send_notification = true
-  end
-end
+require 'journal_listener'
```

Comment:
Not bad xD

Better: Remove the subscribing part from the listener itself and move it explicitly into this initializer.

Something along the lines of

````
OpenProject::Notifications.subscribe('journal_created') do |payload|
  JournalListener.distinguish_journals(payload[:journal], payload[:send_notification])
end
````

Thereby you separate the setup (registering the listener) from the living part of the implementation.

*Additional note:* If I remember correctly there is something to wrap your call into, to ensure that the registration is performed once in production and before every request in development mode. This *might* be neccessary here...

*Addition to the additon:* With "might" I mean, that I have no idea, but it is in the realm of possibility to be true.